### DISCUSSION TITLE
mocking `send(..., { to: 'child' })` ?

### DISCUSSION THREAD

**Discussion by ilovett** [statelyai/xstate#2056] [Q&A]
What is the right way to mock `send` (and `forwardTo`) - just to strip out the `{ to: 'child' }` so I can isolate testing the parent machine only?

Heres what I currently have:

```ts
const mockSend = jest.fn().mockImplementation((context, event) => null);

jest.mock('xstate', () => ({
  ...(jest.requireActual('xstate') as object),
  send: () => (...args: any[]) => mockSend(...args),
  forwardTo: jest.fn().mockImplementation(() => () => null),
}));
```

Problem with this, is that I found mocking `send` seems to break delayed transitions / timeouts queued for simulated clock.

I do see on the interperter `sendTo` -- is it possible to mock this only?

**Comment by davidkpiano** [comment]
Try not to mock anything (using `jest.mock`, etc.). Instead, implement a dummy service:

```js
const machine = createMachine({ ... }, {
  services: {
    child: () => { ... }
  }
});
```

**Reply by ilovett** [reply]
@davidkpiano thanks, something is not clicking for me... I have named actions that create child machines and store the ref on the context.  In some other actions, I use `send(..., { to: 'childId' })` -- I am not sure how overriding the services will override those refs or ids ?

**Reply by ilovett** [reply]
```js
actions: {
  createChildMachine: assign({
    childRef: (context, event) => {
      return spawn(
        createChildMachine(...),
        { name: 'childId' }
      );
    },
  })
}
```
```js
// test overrides to not actually create children
parentMachine.withConfig({
  actions: {
    createChildMachine: jest.fn(),
  }
})
```

**Reply by davidkpiano** [reply]
They won't (yet), named actors aren't yet supported for `spawn(...)`.
